fourohtwo.js

```diff
@@ -104,14 +105,29 @@ const app = express();
 const lsatRouter = express.Router();
 const appRouter = express.Router();
 
+let newInvoice = null;
+
 appRouter.get("/", async function (req, res) {
-  const invoice = await lnd.makeInvoice({ amount: 100, memo: "a402" });
-  res.render("index", { invoice: invoice.data, headers: req.headers, user: req.user });
+  res.render("index", { newInvoice, headers: req.headers, user: req.user });
+  if (newInvoice) {
+    newInvoice = null
+  }
 });
 
+// invoice calls
+
 appRouter.post("/invoice", async function (req, res) {
-  const invoice = await lnd.makeInvoice({ amount: 100, memo: "a402" });
-  res.json({ payment_request: invoice.data.payment_request });
+  if (req.body.amount && req.body.name) {
+    const invoice = await lnd.makeInvoice({ amount: req.body.amount, memo: "a402" });
+    const newInvoiceObj = {
+      id: uuid.v4(),
+      name: req.body.name,
+      amount: req.body.amount,
+      data: invoice.data
+    }
+    newInvoice = newInvoiceObj;
```

Comment:
You should not use global variables. 
If I understand your code correct, then here a global variable is defined with an invoice object. Then the user is redirected to `/` on `/` a template is rendered that uses that global variable and then that global variable is set to null again. 

But think about what happens if two or more users use the app at the same time. Then that one global variable is shared among all users and they overwrite each other. 

I think generally no global variables should be used. HTTP is stateless and when build web apps we need to respect that.